Add some tests for SVG resource invalidation

heycam opened the pull request:
#### f22c7781ad7df4488303c83980ae1ae4c5391cfc
<pre>
Add some tests for SVG resource invalidation
<a href="https://bugs.webkit.org/show_bug.cgi?id=242497">https://bugs.webkit.org/show_bug.cgi?id=242497</a>

Reviewed by Nikolas Zimmermann.

The filter and mask tests are currently failing in their second
sub-tests, where the URL reference is changed from one resource to
another, but no update happens.

* LayoutTests/svg/resource-invalidation/clip-path-complex-resource-invalidation-expected.html: Added.
* LayoutTests/svg/resource-invalidation/clip-path-complex-resource-invalidation.html: Added.
* LayoutTests/svg/resource-invalidation/clip-path-resource-invalidation-expected.html: Added.
* LayoutTests/svg/resource-invalidation/clip-path-resource-invalidation.html: Added.
* LayoutTests/svg/resource-invalidation/filter-resource-invalidation-expected.html: Added.
* LayoutTests/svg/resource-invalidation/filter-resource-invalidation.html: Added.
* LayoutTests/svg/resource-invalidation/gradient-resource-invalidation-expected.html: Added.
* LayoutTests/svg/resource-invalidation/gradient-resource-invalidation.html: Added.
* LayoutTests/svg/resource-invalidation/mask-resource-invalidation-expected.html: Added.
* LayoutTests/svg/resource-invalidation/mask-resource-invalidation.html: Added.
* LayoutTests/svg/resource-invalidation/pattern-resource-invalidation-expected.html: Added.
* LayoutTests/svg/resource-invalidation/pattern-resource-invalidation.html: Added.

Canonical link: <a href="https://commits.webkit.org/252476@main">https://commits.webkit.org/252476@main</a>
</pre>

nikolaszimmermann reviewed (changes requested):
Great, thanks for the tests!
I would very much prefer if you could keep the svg/ tests in svg/, not fast/svg. Much easier for SVG development to run only svg/ tests during main development cycle, and the full layout test run more seldom.

nikolaszimmermann reviewed (commented):
Great to see new reftests covering invalidation, thanks Cameron!

heycam committed:
Add some tests for SVG resource invalidation
https://bugs.webkit.org/show_bug.cgi?id=242497

Reviewed by Nikolas Zimmermann.

The filter and mask tests are currently failing in their second
sub-tests, where the URL reference is changed from one resource to
another, but no update happens.

* LayoutTests/svg/resource-invalidation/clip-path-complex-resource-invalidation-expected.html: Added.
* LayoutTests/svg/resource-invalidation/clip-path-complex-resource-invalidation.html: Added.
* LayoutTests/svg/resource-invalidation/clip-path-resource-invalidation-expected.html: Added.
* LayoutTests/svg/resource-invalidation/clip-path-resource-invalidation.html: Added.
* LayoutTests/svg/resource-invalidation/filter-resource-invalidation-expected.html: Added.
* LayoutTests/svg/resource-invalidation/filter-resource-invalidation.html: Added.
* LayoutTests/svg/resource-invalidation/gradient-resource-invalidation-expected.html: Added.
* LayoutTests/svg/resource-invalidation/gradient-resource-invalidation.html: Added.
* LayoutTests/svg/resource-invalidation/mask-resource-invalidation-expected.html: Added.
* LayoutTests/svg/resource-invalidation/mask-resource-invalidation.html: Added.
* LayoutTests/svg/resource-invalidation/pattern-resource-invalidation-expected.html: Added.
* LayoutTests/svg/resource-invalidation/pattern-resource-invalidation.html: Added.

Canonical link: https://commits.webkit.org/252476@main
